environments/example-credentials/services/scoringmp/overlays/appsody-init-containers.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: openliberty.io/v1beta1
+kind: OpenLibertyApplication
+metadata:
+  name: scoring-mp
+spec:
+  initContainers:
+  - name: init-bootstrap-properties
+    image: docker.io/busybox:1.32
+    imagePullPolicy: "Always"
+    resources:
+      requests:
+        cpu: 250m
+        memory: 256Mi
+    command:
+      - /bin/sh
+      - -c
+      - |
+        echo "mp.messaging.connector.liberty-kafka.bootstrap.servers=${KAFKA_BROKERS}" > tmpfile
+        echo "mp.messaging.connector.liberty-kafka.security.protocol=SASL_SSL" >> tmpfile
+        echo "mp.messaging.connector.liberty-kafka.ssl.protocol=TLSv1.2" >> tmpfile
+        echo "mp.messaging.connector.liberty-kafka.sasl.mechanism=PLAIN" >> tmpfile
+        echo "mp.messaging.connector.liberty-kafka.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username=\"${KAFKA_USER}\" password=\"${KAFKA_PASSWORD}\";" >> tmpfile
+        cp tmpfile /config/bootstrap.properties
```

Comment:
We have another microservice based on this stack + using microprofile reactive messaging, which is able to pick up these properties as env vars.  Although the bootstrap.servers is injected in a Service Binding request, it's basically an uppercase and dots-to-underscores translation of each property to an env var:
https://github.com/ibm-icpa-coffeeshop/coffeeshop-ui/blob/master/config/service-binding-request.yaml

The env vars are apparently only picked up if they also have defaults specified in `microprofile-config.properties`: https://github.com/OpenLiberty/open-liberty/issues/10575
- I see this was cited as the original reason for specifying the values in bootstrap.properties, but sounds like they could instead be defaulted in microprofile-config.properties and then overridden using env vars.  I haven't tried this though, and the sasl.jaas.config property looks like it could be exciting, but could be worth a shot to clean this up.

Reply:
Yep, that issue was opened when we were originally trying to work with scoring-mp in a Reactive Messaging model. We worked with Andrew to debug it and he was able to help us eliminate the paths that didn't work until we found that path that did (injecting bootstrap.properties entirely). 

The problem comes up when we set defaults for values that Kafka doesn't handle gracefully when they aren't needed, just so we can maybe override them later. Which this also seems like a messy situation to have to build in all possible configuration options you expect people to override... But I may be thinking too much in the general sense and not in a specific per-microservice basis.

I'll do some testing today to see how the defaults & overrides perform.

Reply:
I didn't find a suitable resolution to be able to specify defaults of the SSL truststore properties and then being able to override them later with env vars. Using both `null` and empty values for the properties in `microprofile-config.properties` still had the Reactive MP framework processing them as literals, which throws errors in the non-ES on OCP cases. 

I've refined this a bit more, to split out as much as we can into the `microprofile-config.properties` with values that Kafka provides default values for that we can provide in the built image. These can then be overriden with environment variables. The three remaining properties (`sasl.jaas.config`, `ssl.truststore.location`, and `ssl.truststore.password`) need to be defined in `bootstrap.properties` to take effect.